### ISSUE TITLE
Port our benchmark framework to Bitcoin

### ISSUE THREAD

**Issue by daira** [zcash/zcash#2356]
... in order to compare Zcash with Bitcoin.

**Comment by nathan-at-least** [comment]
The purpose of this ticket is not to merge our benchmarks into upstream, but rather to ensure *we* run those benchmarks against upstream as a comparison.

I assume upstream would not want to merge our benchmarks as-if, mainly because they are in a different framework. I filed #2357 to do that deeper integration with upstream benchmarks as a separate task.

**Comment by nathan-at-least** [comment]
I'm blocked on merges, so I will start on this.

**Comment by nathan-at-least** [comment]
Just started this WIP branch for benchmarking against Bitcoin v0.11.2: https://github.com/zcash/bitcoin/tree/zcash-2356-v0.11.2

**Comment by nathan-at-least** [comment]
Current status: I've copypasta'd `zcbenchmarks.cpp` and the rpcwallet method, then ripped out everything except `sleep` and `validatelargetx`, and I'm currently downgrading `C++11` code in the benchmarking function.

**Comment by nathan-at-least** [comment]
Bumped out of 1.0.9 due to lack of time.
